pilot/pkg/serviceregistry/kube/controller/ambient/policies.go

```diff
@@ -59,6 +61,15 @@ func WaypointPolicyStatusCollection(
 				reason := "unknown"
 				bound := false
 				switch target.GetKind() {
+				case gvk.GatewayClass_v1.Kind:
+					fetchedGatewayClasses := krt.Fetch(ctx, gatewayClasses, krt.FilterKey(target.GetName()))
```

Comment:
Should we enforce that it's bound to a waypoint gatewayclass?

Reply:
I think that would be a good idea. I'll add a check that the controller name of the gatewayclass is `istio.io/mesh-controller`. Are there any other valid controller names for waypoints?

Reply:
I think waypoint is the only one, thanks!